head_ref/include/head_ref/HeadReference.h

```diff
@@ -83,18 +85,21 @@ class HeadReference
          */
         void publishReferences(head_ref_msgs::HeadReferenceGoal& goal);
 
-        HeadReferenceActionServer* as_;
-        std::vector < HeadReferenceActionServer::GoalHandle > goal_handles_;
+        std::unique_ptr<HeadReferenceActionServer> as_;
+        std::vector<HeadReferenceActionServer::GoalHandle> goal_handles_;
 
         bool float_topics_;
         ros::Publisher head_pub_, pan_pub_, tilt_pub_, marker_pub_;
         ros::Subscriber measurement_sub_;
 
-        tf::TransformListener* tf_listener_;
+        std::unique_ptr<tf::TransformListener> tf_listener_;
 
         double current_pan_, current_tilt_, goal_error_tolerance_;
         
         std::string tf_prefix_;
+        std::string frame_mount_;
```

Comment:
* It's good to make things less hardcoded. To me, however, it's not entirely clear how these are used (I assume before, between and after pan and tilt joints). Furthermore, hidden assumptions are important here: I can imagine that if you would change the orientation of one the frames the entire algorithm would malfunction. Therefore, you could argue that we should either have stuff hardcoded so that users cannot screw it up or to make it really generic. The big question here is if we combine the best or the worst of both worlds.
* As a minor point (but that might be due to the total design): I'm not too fond of this many member variables. You could put them in a separate data structure but to really improve this a more elaborate overhaul would be necessary

Reply:
True, I am pretty sure that the frames should have a specific orientation. I don't have a solution yet to make the orientation of the frames configurable.

I think this is an improvement. If you disagree, I just revert this and we work on this later.